java/ql/src/semmle/code/java/frameworks/guava/Guava.qll

```diff
@@ -0,0 +1,34 @@
+/**
+ * Definitions for tracking taint steps throught the Guava framework.
+ */
+
+import java
+private import Strings
+private import Splitter
+private import Joiner
+
+/**
+ * A method in the guava framework that propegates taint.
+ */
+abstract class GuavaTaintPropagationMethod extends Method {
+  /**
+   * Holds if this method propagates taint between the given source and sink.
+   * `src` and `sink` are indicies of arguments to this method, or -1 to represent the qualifier.
+   * `sink` can also be -2 to represent the return value.
+   */
```

Comment:
What woud you suggest otherwise? The -1 as qualifier convention isn't too uncommon elsewhere, and having all the logic in one predictate makes implementation and usage a bit easier. 
A possible alternative could be to implement define the 5 predicates for `qualifierToMethod`, `argToMethod`, etc, mirroring the predicates of `TaintTrackingUtils`.

Reply:
I think using `-1` got the qualifier is fine - as you note, we already do that in several other places, and the qualifier is essentially just another argument, so it's nice for it to have an index.  I'd drop the `-2` magic constant, though, and since that one also has different limitations on possible flow (it can only be an exit point, never an entry), it means that we'll only need a total of two predicates to cover all combinations.